src/uct/ib/mlx5/dv/ib_mlx5_ifc.h

```diff
@@ -1505,4 +1520,42 @@ enum {
     UCT_IB_MLX5_EVENT_TYPE_SRQ_LAST_WQE       = 0x13
 };
 
+struct uct_ib_mlx5_general_obj_out_cmd_hdr_bits {
+    uint8_t         status[0x8];
+    uint8_t         reserved_at_8[0x18];
+
+    uint8_t         syndrome[0x20];
+
+    uint8_t         obj_id[0x20];
+
+    uint8_t         reserved_at_60[0x20];
+};
+
+struct uct_ib_mlx5_general_obj_in_cmd_hdr_bits {
+    uint8_t         opcode[0x10];
+    uint8_t         uid[0x10];
+
+    uint8_t         reserved_at_20[0x10];
+    uint8_t         obj_type[0x10];
+
+    uint8_t         obj_id[0x20];
+
+    uint8_t         reserved_at_60[0x3];
+    uint8_t         log_obj_range[0x5];
+    uint8_t         reserved_at_68[0x18];
+};
+
+struct uct_ib_mlx5_reserved_qpn_bits {
+    uint8_t         reserved_at_0[0x80];
+};
+
+struct uct_ib_mlx5_create_reserved_qpn_in_bits {
+    struct uct_ib_mlx5_general_obj_in_cmd_hdr_bits  hdr;
+    struct uct_ib_mlx5_modify_qp_in_bits            qpns;
+};
+
+enum {
+    UCT_IB_MLX5_OBJ_TYPE_RESERVED_QPN = 0x002C,
```

Comment:
@evgeny-leksikov It can be used to indicate the bit of cap, and it's also used as the type specified as  input to create the RESERVED_QPN obj. So it's not only used for cap.
```
    attr = UCT_IB_MLX5DV_ADDR_OF(create_reserved_qpn_in, in, hdr);
    UCT_IB_MLX5DV_SET(general_obj_in_cmd_hdr,
         attr, opcode, UCT_IB_MLX5_CMD_OP_CREATE_GENERAL_OBJECT);
    UCT_IB_MLX5DV_SET(general_obj_in_cmd_hdr,
         attr, obj_type, UCT_IB_MLX5_OBJ_TYPE_RESERVED_QPN);
    UCT_IB_MLX5DV_SET(general_obj_in_cmd_hdr,
         attr, log_obj_range, ctx->log_reserved_qpn_granularity);

    blk->obj = mlx5dv_devx_obj_create(verbs, in, sizeof(in),
                                      out, sizeof(out));```

Reply:
maybe better to have 2 different constants for different cases

Reply:
@evgeny-leksikov In PRM they are shared with the same value and naming. UCT_IB_MLX5_OBJ_TYPE_RESERVED_QPN is used by cap but it's not cap itself, it just indicate the bit number of the corresponding obj type  in the cap.

Reply:
i see your point but it's a bit confusing to me, @yosefe WDYT?

Reply:
IMO, it should be same name. 
"obj_types" is a bitmap of supported object types, in `(ctx->general_obj_types_caps & (1ULL << UCT_IB_MLX5_OBJ_TYPE_RESERVED_QPN))`
BTW. better use UCS_BIT() than 1ULL<<

Reply:
Changed to use UCX_BIT()